packages/edge-login-ui-rn/src/components/scenes/ChangePinScene.tsx

```diff
@@ -0,0 +1,183 @@
+import { EdgeAccount } from 'edge-core-js'
+import * as React from 'react'
+import { Keyboard, ScrollView, View } from 'react-native'
+import { cacheStyles } from 'react-native-patina'
+
+import { completeResecure } from '../../actions/LoginCompleteActions'
+import { onComplete } from '../../actions/WorkflowActions'
+import s from '../../common/locales/strings'
+import { useScrollToEnd } from '../../hooks/useScrollToEnd'
+import { Dispatch, RootState } from '../../types/ReduxTypes'
+import { logEvent } from '../../util/analytics'
+import { getAccount } from '../../util/selectors'
+import { ButtonsModal } from '../modals/ButtonsModal'
+import { Airship, showError } from '../services/AirshipInstance'
+import { connect } from '../services/ReduxStore'
+import { Theme, ThemeProps, withTheme } from '../services/ThemeContext'
+import { DigitInput, MAX_PIN_LENGTH } from '../themed/DigitInput'
+import { EdgeText } from '../themed/EdgeText'
+import { Fade } from '../themed/Fade'
+import { MainButton } from '../themed/MainButton'
+import { ThemedScene } from '../themed/ThemedScene'
+
+interface OwnProps {
+  newAccount: boolean
+}
+
+interface StateProps {
+  account?: EdgeAccount
+  pin: string
+  pinError: string
+  pinErrorMessage: string | null
+}
+
+interface DispatchProps {
+  onDone: () => void
+  onBack: () => void
+}
+
+type Props = OwnProps & StateProps & DispatchProps & ThemeProps
+
+const ChangePinSceneComponent = ({
```

Comment:
Instead of writing a component that accepts the 'newAccount' prop and has a bunch of conditions, it's better if you just passed in every conditioned prop separately. 
That way the component can be a simple visual component without any logic.

Then you can create 3 "wrapping" components like you already did at the bottom that each passes the correct props.

That way you won't need to even have the 'newAccount' prop and each scene would have a unique component.